File: 2/code.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        node = ListNode(None)
        retNode = node
        carry = 0
        dig = 0
        flag = 0
        while l1 is not None and l2 is not None:
            if flag != 0:
                node.next = ListNode(None)
                node = node.next
            tmp = l1.val + l2.val + carry
            carry = tmp / 10
            dig = tmp % 10
            node.val = dig
            l1 = l1.next
            l2 = l2.next
            flag += 1
        while l1 is not None:
            if flag != 0:
                node.next = ListNode(None)
                node = node.next
            tmp = l1.val + carry
            carry = tmp / 10
            dig = tmp % 10
            node.val = dig
            l1 = l1.next
            flag += 1
        while l2 is not None:
            if flag != 0:
                node.next = ListNode(None)
                node = node.next
            tmp = l2.val + carry
            carry = tmp / 10
            dig = tmp % 10
            node.val = dig
            l2 = l2.next
            flag += 1
        if carry != 0:
            node.next = ListNode(None)
            node = node.next
            node.val = carry
        return retNode

File: 2/code.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var addTwoNumbers = function(l1, l2) {
    var node = new ListNode(),
        retNode = node,
        carry = 0,
        dig = 0,
        tmp, flag;
    function div(num1, num2) {
        num1 = Math.round(num1);
        num2 = Math.round(num2);
        var result = num1 / num2;
        if (result >= 0){
            result = Math.floor(result);
        } else {
            result = Math.ceil(result);
        }
        return result;
    }
    flag = 0;
    while (l1 !== null && l2 !== null) {
        if (flag !== 0) {
            node.next = new ListNode();
            node = node.next;
        }
        tmp = l1.val + l2.val + carry;
        carry = div(tmp, 10);
        dig = tmp % 10;
        node.val = dig;
        l1 = l1.next;
        l2 = l2.next;
        flag += 1;
    }
    while (l1 !== null) {
        if (flag !== 0) {
            node.next = new ListNode();
            node = node.next;
        }
        tmp = l1.val + carry;
        carry = div(tmp, 10);
        dig = tmp % 10;
        node.val = dig;
        l1 = l1.next;
        flag += 1;
    }
    while (l2 !== null) {
        if (flag !== 0) {
            node.next = new ListNode();
            node = node.next;
        }
        tmp = l2.val + carry;
        carry = div(tmp, 10);
        dig = tmp % 10;
        node.val = dig;
        l2 = l2.next;
        flag += 1;
    }
    if (carry !== 0) {
        node.next = new ListNode();
        node = node.next;
        node.val = carry;
    }
    return retNode;
};